webapp/platform/client/src/websocket.ts

```diff
@@ -71,7 +86,9 @@ export default class WebSocketClient {
     private serverHostname: string | null;
     private postedAck: boolean;
 
-    constructor() {
+    private reconnectTimeout: ReturnType<typeof setTimeout> | null;
+
+    constructor(config?: Partial<WebSocketClientConfig>) {
```

Comment:
Why are we adding a possibility to add a config, if we are not using that possibility in the code?

Reply:
I added the config to be used by the unit tests - it's used to specify specific values for the timeouts, and to allow mocking the internal websocket. The config _could_ be used by production code, but that was not my intention 🙂

Reply:
Short answer: :+1: LGTM

Ignorable rant:
I have mixed feelings about adding extra things just for tests. They increase the interface, and people may start using it :P On top of that, it increases the cognitive load, since now is not that we are using the same constant everywhere. We are using whatever comes in the config. So who is passing the config? Er... no one. We always use the default :P

In general, I would love to see the tests not relying on anything extra. If that is not possible, I like things exported for tests to be really revealing (like an export called "exportForTests").

Reply:
I have another PR coming to add a client side websocket ping to check for broken connections - I'll see if I can clean this up a bit on that PR. I'll merge this one as-is to keep things moving 👍